capella_model_explorer/__main__.py

```diff
@@ -145,143 +144,151 @@ def build_css(*, watch: bool) -> subprocess.Popen | None:
     ]
     if watch:
         tailwind_cmd.append("--watch")
-        print(shlex.join(tailwind_cmd))
+        logger.info(shlex.join(tailwind_cmd))
         return subprocess.Popen(tailwind_cmd)
-    print(shlex.join(tailwind_cmd))
+
+    logger.info(shlex.join(tailwind_cmd))
     subprocess.check_call(tailwind_cmd)
     return None
 
 
-def build_image():
-    if not (pathlib.Path.cwd() / "capella_model_explorer").is_dir():
-        raise SystemExit("Run this command from the root of the project.")
-    docker = _find_exe("docker")
-    cmd = [
-        docker,
-        "build",
-        "--no-cache",
-        "-t",
-        f"capella-model-explorer:{capella_model_explorer.__version__}",
-        ".",
-    ]
-    print(shlex.join(cmd))
-    subprocess.check_call(cmd)
-
-
-def pre_commit_setup():
-    pre_commit = _find_exe("pre-commit")
-    print("Installing tools needed for pre-commit hooks...")
-    _install_npm_pkgs()
-    print("Installing pre-commit hooks...")
-    subprocess.check_call([pre_commit, "install-hooks"])
-    subprocess.check_call([pre_commit, "install"])
-
-
 def _find_exe(name: str) -> str:
     path = shutil.which(name)
     if path is None:
         raise SystemExit(f"Cannot find {name!r}, install it and try again")
     return path
 
 
-def main():
-    parser = argparse.ArgumentParser(
-        description=(
-            "CLI for managing the application. Every command"
-            " has its own `--help` option."
+@click.group()
+def main() -> None:
+    """Command line interface to control the application."""
+
+
+@main.command()
+@click.option(
+    "--container",
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Launch as a Docker container.",
+)
+@click.option(
+    "--dev",
+    is_flag=True,
+    show_default=True,
+    help="Launch in development mode with full auto-reload."
+    " This option cannot be used together with --container.",
+)
+@click.option(
+    "--host",
+    envvar="CME_HOST",
+    default=c.Defaults.host,
+    show_default=True,
+    help="The hostname or IP address to bind to."
+    " Ignored when running with '--container'.",
+)
+@click.option(
+    "--port",
+    envvar="CME_PORT",
+    default=c.Defaults.port,
+    show_default=True,
+    help="The port to listen on.",
+)
+@click.option(
+    "--model",
+    envvar="CME_MODEL",
+    default=c.Defaults.model,
+    show_default=True,
+    help="The Capella model to load (file, URL or JSON string). ",
+)
+@click.option(
+    "--templates-dir",
+    envvar="CME_TEMPLATES_DIR",
+    default=str(c.Defaults.templates_dir.resolve()),
+    show_default=True,
+    help="The directory containing the templates. ",
```

Comment:
Here is also a blank that can be removed.

Reply:
Yes. That's why I wrote "some of the help texts", not "this help text". I figured that a dev as experienced as Ernst would be able to find the other occurences without me explicitly marking them :)